src/algorithms/chain_items.hpp

```diff
@@ -381,14 +381,33 @@ ostream& operator<<(ostream& out, const TracedScore& value);
  */
 void sort_anchor_indexes(const std::vector<Anchor>& items, std::vector<size_t>& indexes);
 
+struct transition_info {
+    // Index of the source anchor within the list of anchors
+    size_t from_anchor;
+    // Index of the destination anchor
+    size_t to_anchor;
+    // Distance between anchors in the graph
+    size_t graph_distance;
+    // Distance between anchors in the read (defaults to max if not set)
+    size_t read_distance = std::numeric_limits<size_t>::max();
+
+    // Constructor without read distance
+    transition_info(size_t from, size_t to, size_t graph_dist)
+        : from_anchor(from), to_anchor(to), graph_distance(graph_dist) {}
+    
+    // Constructor with read distance
+    transition_info(size_t from, size_t to, size_t graph_dist, size_t read_dist)
+        : from_anchor(from), to_anchor(to), graph_distance(graph_dist), read_distance(read_dist) {}
```

Comment:
These could be one signature with `size_t read_dist = std::numeric_limits<size_t>::max()`.

Also, I think they might need to be `inline` if the bodies are in the HPP here, even if the bodies are empty, or they will violate the one-definition rule.